sympy/physics/mechanics/kane.py

```diff
@@ -15,49 +13,99 @@
 from sympy.utilities.exceptions import SymPyDeprecationWarning
 from sympy.utilities.iterables import iterable
 
+__all__ = ['KanesMethod']
+
+
+def _none_handler(x):
+    """Returns and empty matrix if x is False."""
+    if x:
+        return Matrix(x)
+    else:
+        return Matrix()
+
 
 class KanesMethod(object):
-    """Kane's method object.
+    """This class is used automatically form the equations of motion of a
+    multibody system using Kane's method [Kane1985]_.
+
+    The dynamic equations of motion are provided in Kane's form::
+
+        fr + frstar = 0
+
+    in the factored implicit form::
+
+        mass_matrix * accelerations = forcing
+
+    or in the explicit form::
 
-    This object is used to do the "book-keeping" as you go through and form
-    equations of motion in the way Kane presents in:
-    Kane, T., Levinson, D. Dynamics Theory and Applications. 1985 McGraw-Hill
+        accelerations = mass_matrix.inv() * forcing
 
-    The attributes are for equations in the form [M] udot = forcing.
+    where fr, frstar, the mass matrix, and the right hand side vector are
+    available as class attributes after Kane's equations are formed.
 
     Attributes
     ==========
 
-    q, u : Matrix
-        Matrices of the generalized coordinates and speeds
-    bodylist : iterable
-        Iterable of Point and RigidBody objects in the system.
-    forcelist : iterable
-        Iterable of (Point, vector) or (ReferenceFrame, vector) tuples
-        describing the forces on the system.
+    inertial_frame : ReferenceFrame
+        The inertial reference frame the system is defined with respect to.
+    n, num_coordinates, num_speeds : integer
+        The total number of generalized coordinates and speeds.
+    q, coordinates : Matrix, shape(n, 1)
+        The generalized coordinates. These are ordered as [q_s, q_r]^T.
```

Comment:
Yes you are correct. This needs some distinction. n should just be the number of generalized speeds I believe. I need to look back at kane's book to see how we handles this.